test/e2e/operator/deployment_api.go

```diff
@@ -775,6 +776,70 @@ var _ = deploy.DescribeForSome("API", func(d *deploy.Deployment) bool {
 			}
 		})
 	})
+
+	Context("conversion", func() {
+		getAlphaDeployment := func(name string) alphaapi.Deployment {
+			return alphaapi.Deployment{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: name,
+				},
+				Spec: alphaapi.DeploymentSpec{
+					DeploymentSpec: base.DeploymentSpec{
+						Image: dummyImage,
+					},
+				},
+			}
+		}
+		It("with default values", func() {
+			if d.HasOLM {
+				Skip("Conversion webhook is not supported by OLM")
+			}
+			alphaDep := getAlphaDeployment("alpha-default-values")
+			deploy.CreateAlphaDeploymentCR(f, alphaDep)
+
+			deployment := deploy.GetDeploymentCR(f, alphaDep.Name)
+
+			defer deploy.DeleteDeploymentCR(f, deployment.Name)
+
+			validateDriver(deployment, true)
+		})
+		It("with explicit values", func() {
+			if d.HasOLM {
+				Skip("Conversion webhook is not supported by OLM")
+			}
+			alphaDep := getAlphaDeployment("alpha-explicit-values")
+			alphaDep.Spec.NodeResources = &corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("10m"),
+					corev1.ResourceMemory: resource.MustParse("25Mi"),
+				},
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("100m"),
+					corev1.ResourceMemory: resource.MustParse("50Mi"),
+				},
+			}
+			alphaDep.Spec.ControllerResources = &corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("20m"),
+					corev1.ResourceMemory: resource.MustParse("50Mi"),
+				},
+				Limits: corev1.ResourceList{
+					corev1.ResourceCPU:    resource.MustParse("200m"),
+					corev1.ResourceMemory: resource.MustParse("100Mi"),
+				},
+			}
+			deploy.CreateAlphaDeploymentCR(f, alphaDep)
+
+			deployment := deploy.GetDeploymentCR(f, alphaDep.Name)
+
+			Expect(deployment.Spec.NodeDriverResources).Should(BeEquivalentTo(alphaDep.Spec.NodeResources), "node driver resources")
+			Expect(deployment.Spec.ControllerDriverResources).Should(BeEquivalentTo(alphaDep.Spec.ControllerResources), "controller driver resources")
+
+			defer deploy.DeleteDeploymentCR(f, deployment.Name)
+
+			validateDriver(deployment, true)
+		})
+	})
```

Comment:
According to the conversion functions above, conversion from beta to alpha is supported. This is not tested here, is it?

Reply:
Though the conversion code is available, the operator does not reconcile `v1alpha1` CRs, and thus no test to cover that.

Reply:
Extended the tests to make sure that the `v1beta1` ->``v1alpha1` conversion works:
 - create `v1alpha1` object
 - retrieve back the `v1alpha1` revision and make sure that both the `spec` fields match.
 - retrieve the `v1beta1` revision for the same object and check that the `status` matches with the `v1alpah1` status.

Reply:
"conversion with explicit values" fails: https://cloudnative-k8sci.southcentralus.cloudapp.azure.com/blue/organizations/jenkins/pmem-csi/detail/PR-826/15/tests

Reply:
> Extended the tests to make sure that the v1beta1 ->``v1alpha1` conversion works:

This isn't really converting something that was created as v1beta1, is it?

Reply:
> > Extended the tests to make sure that the v1beta1 ->``v1alpha1` conversion works:
> 
> This isn't really converting something that was created as v1beta1, is it?

The test creates a `v1alpha1` object, which actually stored as the `v1beta1` by the API server after the conversion. Then the test retrieves back both `v1alpah1` and `v1beta1` versions for the same CR, and checks if both have expected fields.